cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -144,32 +143,21 @@ func doSystemdMountImpl(what, where string, opts *systemdMountOptions) error {
 		args = append(args, "--options="+strings.Join(options, ","))
 	}
 
-	// note that we do not currently parse any output from systemd-mount, but if
-	// we ever do, take special care surrounding the debug output that systemd
-	// outputs with the "debug" kernel command line present (or equivalently the
-	// SYSTEMD_LOG_LEVEL=debug env var) which will add lots of additional output
-	// to stderr from systemd commands
-	out, err := exec.Command("systemd-mount", args...).CombinedOutput()
-	if err != nil {
-		return osutil.OutputErr(out, err)
-	}
-
 	// if it should survive pivot_root() then we need to add overrides for this
 	// unit to /run/systemd units
 	if !opts.Ephemeral {
-		// to survive the pivot_root, we need to make the mount units depend on
-		// all of the various initrd special targets by adding runtime conf
-		// files there
+		args = append(args, "--property=DefaultDependencies=no")
+
+		// to survive the pivot_root, mounts need to be "wanted" by
+		// initrd-switch-root.target directly or indirectly. The
+		// proper target to place them in is initrd-fs.target
 		// note we could do this statically in the initramfs main filesystem
 		// layout, but that means that changes to snap-bootstrap would block on
 		// waiting for those files to be added before things works here, this is
 		// a more flexible strategy that puts snap-bootstrap in control
 		overrideContent := []byte(fmt.Sprintf(unitFileDependOverride, unitName))
 		for _, initrdUnit := range []string{
```

Comment:
if the only conf file we need to write to is `initrd-fs.target`, then this doesn't need to be a for loop anymore

Reply:
I have changed that.